test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs

```diff
@@ -0,0 +1,114 @@
+
+            // Act
+            var actualLocalizedString = htmlLocalizer["Hello"];
+
+            // Assert
+            Assert.Equal(localizedString, actualLocalizedString);
+        }
+
+        [Fact]
+        public void HtmlLocalizer_UseIndexerWithArguments_ReturnsLocalizedString()
+        {
+            // Arrange
+            var localizedString = new LocalizedString("Hello", "Bonjour test");
+
+            var stringLocalizer = new Mock<IStringLocalizer>();
+            stringLocalizer.Setup(s => s["Hello", "test"]).Returns(localizedString);
+
+            var htmlLocalizer = new HtmlLocalizer(stringLocalizer.Object, new CommonTestEncoder());
+
+            // Act
+            var actualLocalizedString = htmlLocalizer["Hello", "test"];
+
+            // Assert
+            Assert.Equal(localizedString, actualLocalizedString);
+        }
+
+        public static IEnumerable<object[]> HtmlData
+        {
+            get
+            {
+                yield return new object[] { "Bonjour {0} {{{{ }}", new object[] { "test" }, "Bonjour HtmlEncode[[test]] {{ }" };
+                yield return new object[] { "Bonjour {{0}}", new object[] { "{0}" }, "Bonjour {0}" };
+                yield return new object[] { "Bonjour {0:x}", new object[] { 10 }, "Bonjour HtmlEncode[[a]]" };
+                yield return new object[] { "Bonjour {0:x}}}", new object[] { 10 }, "Bonjour x}" };
+                yield return new object[] { "Bonjour {{0:x}}", new object[] { 10 }, "Bonjour {0:x}" };
+                yield return new object[] { "{{ Bonjour {{{0:x}}}", new object[] { 10 }, "{ Bonjour {x}" };
+                yield return new object[] { "}} Bonjour {{{0:x}}}", new object[] { 10 }, "} Bonjour {x}" };
+                yield return new object[] { "}} Bonjour", new object[] { }, "} Bonjour" };
+                yield return new object[] { "{{ {0} }}", new object[] { 10 }, "{ HtmlEncode[[10]] }" };
+                yield return new object[] {
+                    "Bonjour {{{0:x}}} {1:yyyy}",
+                    new object[] { 10, new DateTime(2015, 10, 10) },
+                    "Bonjour {x} HtmlEncode[[2015]]"
+                };
+                yield return new object[] {
+                    "Bonjour {{{0:x}}} Bienvenue {{1:yyyy}}",
+                    new object[] { 10, new DateTime(2015, 10, 10) },
+                    "Bonjour {x} Bienvenue {1:yyyy}"
+                };
+                yield return new object[] {
+                    "Bonjour {0,6} Bienvenue {{1:yyyy}}",
+                    new object[] { 10, new DateTime(2015, 10, 10) },
+                    "Bonjour HtmlEncode[[    10]] Bienvenue {1:yyyy}"
+                };
+                yield return new object[] {
+                    "{0:'{{characters that should be escaped}}b'###'b'}",
+                    new object[] { 10 },
+                    "{characters that should be escaped}b10b"
```

Comment:
Please file a but about lack of HTML encoding in this case. (I meant "should be encoded" and they were successfully escaped :smile_cat:) Since literal characters within format tokens are uncommon, probably fine to HTML encode them.

(Another choice would be to `throw` some `Exception` if a format token contains a backslash, single, or double quote. But that wouldn't cover other characters in the format token that are copied directly to the result.)